python/ray/air/_internal/usage.py

```diff
@@ -115,3 +117,29 @@ def tag_scheduler(scheduler: "TrialScheduler"):
     assert isinstance(scheduler, TrialScheduler)
     scheduler_name = _find_class_name(scheduler, "ray.tune.schedulers", TUNE_SCHEDULERS)
     record_extra_usage_tag(TagKey.TUNE_SCHEDULER, scheduler_name)
+
+
+def tag_env_vars() -> bool:
+    """Records environment variable usage.
+
+    Returns:
+        bool: True if at least one environment var is supplied by the user.
+    """
+    from ray.air.constants import AIR_ENV_VARS
+    from ray.tune.constants import TUNE_ENV_VARS
+    from ray.train.constants import TRAIN_ENV_VARS
+
+    all_env_vars = sorted(set().union(AIR_ENV_VARS, TUNE_ENV_VARS, TRAIN_ENV_VARS))
+
+    user_supplied_env_vars = []
+
+    for env_var in all_env_vars:
+        if env_var in os.environ:
+            user_supplied_env_vars.append(env_var)
```

Comment:
Ooc, why not also do set intersection here?

Reply:
Yeah, thats a nicer way!

Reply:
Oh, actually. The user might have way more env variables, so I don't want to incur the cost of converting the env var keys to a set. I'd rather just check for existence and loop through the smaller set of AIR env variables. I'll keep the current way.